src/sagas/nanoContract.js

```diff
@@ -172,21 +171,37 @@ export function* registerNanoContract({ payload }) {
  * @param {string} after Transaction hash to start to get items
  * @param {Object} wallet Wallet instance from redux state
  *
+ * @description
+ * We use `after` because we look at time from present to future, therefore `after` means
+ * "from now on in the future".
+ *
  * @throws {Error} when request code is greater then 399 or when response's success is false
  */
 export async function fetchHistory(ncId, count, after, wallet) {
+  // getNanoContractHistory returns a list of transactions by ascending order,
+  // however its pagination works by descending order.
+  // That's why we ask for new transactions `before` an specific tx.
+  // Pagination: future (newest tx) <- current last tx (before) -> past (oldest tx).
+  // Page order: past (oldest tx) -> future (newest tx).
   /**
    * @type {RawNcTxHistoryResponse} response
    */
-  const response = await ncApi.getNanoContractHistory(ncId, count, after);
+  const response = await ncApi.getNanoContractHistory(
+    ncId,
+    count,
+    /* past */ null,
+    /* future */ after
+  );
   const { success, history: rawHistory } = response;
 
   if (!success) {
     throw new Error('Failed to fetch nano contract history');
   }
-
-  const history = [];
-  for (const rawTx of rawHistory) {
+  // We are interested to produce a list of transactions in descending order
+  // because we want users to see newest txs first.
+  const historyReversed = new Array(rawHistory.length)
+  for (let idx = 0; idx < rawHistory.length; idx += 1) {
```

Comment:
suggestion(non-blocking): You could refactor this to use `Promise.all` so those multiple async requests would run concurrently

Reply:
But if we do this we can't use guarantee the order of execution, right? I need it ordered.

Reply:
You could pass the index to the promise and return it like [index, response], then use it to assembly the array

But it's non-blocking at all, if you think it's not worth the effort, don't do it

Reply:
Ok. I have opened an issue for it: https://github.com/HathorNetwork/hathor-wallet-mobile/issues/514